### ISSUE TITLE
Dev: WAVE Accessibility: Create Project Pages, Navigation Buttons

### ISSUE THREAD

**Issue by ryanfchase** [hackforla/tdm-calculator#2571]
### Overview
We need to fix an accessibility error in the Create Project pages, which are preventing assistive technologies from describing the function of the navigation buttons,

### More Info

The WAVE Tool error, Empty Button, indicates a button is empty or has no value text. This means that screen readers are unable to indicate the function of the button.

In the case of the Create Project navigation buttons, the button has an SVG element as its contents, rather than text. We can fix this by adding the `aria-label` attribute to the button, providing it with descriptive text.

Note: elements with style `display: none` are hidden from assistive technologies.[^1] We can ignore the Empty Button error that is being triggered on hidden navigation buttons. See reference in Screenshots below, "Create Project, Page 1 and Page 4: Empty Button (No Action Required)".

### Action Items
- [x] Download WAVE Tool Chrome Extension (see dropdown: Testing Resources)
  - [x] Run the WAVE Tool on Create Project Page 1
  - [x] Ensure you can see relevant WAVE Errors
    - see dropdown: Create Project Page 1: Empty Button
- [x] Review Dev Resources (see dropdown: NavButton or Button Component Suggested Edits
  - [x] Determine whether you will pursue strategy 1, strategy 2, or come up with a different approach, documenting your reasoning in a comment
- [x] Add attribute `aria-label="project creation next page"` to the nav forward button
- [x] Add attribute `aria-label="project creation previous page"` to the navigation previous button
- [x] Run the WAVE Tool on Create Project Page 1 (again), confirm the error is resolved
  - see dropdown: Create Project, Page 1: No Empty Button Error (For Verification)

#### Staging and Release decks
- [Staging Deck](https://docs.google.com/presentation/d/1crZ3IxqA4hAu3qzD7ns93Ieuqjwh6wyEtuX_46cP-fg/)
   - [Staging slide, [2571](https://docs.google.com/presentation/d/1crZ3IxqA4hAu3qzD7ns93Ieuqjwh6wyEtuX_46cP-fg/edit?slide=id.g3691501d13a_2_27#slide=id.g3691501d13a_2_27)]
- Release Deck
   - [2025-MM-DD Release deck, slide ISSUE NUMBER]

#### Developer Resources

<details><summary>NavButton or Button Component Suggested Edits</summary>
<p>

Either of the following strategies will resolve the Empty Button WAVE error:

Strategy 1: In NavButton.jsx, wrap the `<Button>` Component with an element that has an `aria-label` property. Consider using `navigationDirection` to act as the conditional.
```jsx
// client/src/components/Button/NavButton.jsx
// const NavButton = ({ ...
return (
    {/* ensure <Button> is encapsulated with an element with aria-label */}
    <>
      <Button
        id={id}
        className={clsx(classes.navButton, maybeDisabled, maybeHiddenVisibility)}
        data-testid={id}
        color={color}
        onClick={onClick}
        disabled={isDisabled}
      >
        {navDirection === "previous" ? (
          <MdChevronLeft fontSize="2em" />
        ) : (
          <MdChevronRight fontSize="2em" />
        )}
      </Button>
    </>
  );
```

Strategy 2: In Button.jsx, add an optional prop, `ariaLabel`, and conditionally apply[^2] this attribute to the `<button>` element when present, and modify NavButton.jsx to utilize this prop. Again, consider using `navigationDirection` as the conditional.

```jsx
// client/src/components/Button/Button.jsx
const Button =({ 
  ariaLabel = undefined, // <-- add this here
  // ... other props ...
}) => {
  // ...
return (
    <button
      className={clsx(classes.button, classes[variant], className)}
      onClick={onClick}
      type={type}
      disabled={disabled}
      aria-label={ariaLabel} // <-- add this here, React will omit the attribute on `undefined`
      id={id}
    >
      {children}
    </button>
  );
};
```

Modifying the wrapping NavButton component, as described above...

```jsx
// client/src/components/Button/NavButton.jsx
// const NavButton = ({ ...
return (
      <Button
        id={id}
        className={clsx(classes.navButton, maybeDisabled, maybeHiddenVisibility)}
        data-testid={id}
        color={color}
        onClick={onClick}
        disabled={isDisabled}
        ariaLabel={/* ariaLabel logic goes here */}
      >
        {navDirection === "previous" ? (
          <MdChevronLeft fontSize="2em" />
        ) : (
          <MdChevronRight fontSize="2em" />
        )}
      </Button>
  );
```

You are also welcome to come up with a different strategy as you see fit. Please be sure to document your reasoning, and any research you've done in order to arrive at your solution.

</p>
</details> 

<details><summary>Testing Resources</summary>
<p>

- Dev: https://tdm-dev.azurewebsites.net/calculation/1/0
- Prod: https://tdm.ladot.lacity.org/login
- [WAVE Chrome extension](https://chromewebstore.google.com/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh)

</p>
</details>

<details><summary>WAVE Tool: Empty Button Reference</summary>
<p>

This snippet is taken from clicking on the Empty Button WAVE Icon, and then clicking the Reference button. This information is purely informational <sup>(How ... informative) </sup>

One correction: Per **How to Fix It**, we will be pursuing a different strategy by adding the `aria-label` attribute, since we are using an  SVG for the `<button>`'s content.

```mdx
**What It Means**
A button is empty or has no value text.

**Why It Matters**
When navigating to a button, descriptive text must be presented to screen reader users to indicate the function of the button.

**How to Fix It**
Place text content within the `<button>` element or give the `<input>` element a value attribute.

**The Algorithm... in English**
A `<button>` element is present that contains no text content (or alternative text), or an `<input type="submit">`, `<input type="button">`, or `<input type="reset">` has an empty or missing value attribute.
```

</p>
</details> 

#### Screenshots

<details><summary>Create Project, Page 1: Empty Button</summary>
<p>

Description: A button is empty or has no value text

![Image](https://github.com/user-attachments/assets/941a2b01-f902-414a-a227-927a34217f53)

Same error but with styles turned off

![Image](https://github.com/user-attachments/assets/83fd0eea-7106-4a85-b438-e72a4d17189b)

</p>
</details> 

<details><summary>Create Project, Page 1: No Empty Button Error (For Verification)</summary>
<p>

Description: An aria-label or aria-labelledby attribute is present.

![Image](https://github.com/user-attachments/assets/71bd3cfa-22d9-4c99-89a9-fe6a0a9b8f68)

Same WAVE Icon but also showing Dev Tools Inspector

![Image](https://github.com/user-attachments/assets/7e93fc43-4f18-4efd-86e4-cddda94941ca)

</p>
</details> 

<details><summary>Create Project, Page 1 and Page 4: Empty Button (No Action Required)</summary>
<p>

⚠ Including this info to explicitly call out that we are ignoring this error ⚠ 

See footnote for explanation...[^1]

#### Styles On

![Image](https://github.com/user-attachments/assets/e8402647-0f23-404e-a46f-22275f163286)

#### Styles Off

![Image](https://github.com/user-attachments/assets/1a836250-a44d-4537-ad82-caa5c24dd044)

</p>
</details> 

[^1]: Element with `visibility: hidden;` is hidden from view, but is not removed from the normal flow (i.e., it still takes up the space it normally would). Content is ignored by screen readers Element with `display: none;` is removed from the normal flow and hidden; the space it occupied is collapsed | Content is ignored by screen readers. ([source](https://alistapart.com/article/now-you-see-me/))

[^2]: React intelligently handles attributes that are set to false, null, or undefined, often omitting them from the DOM. ([source](https://truecoderguru.com/blog/conditionally-add-attributes-react-components#using-conditional-props-in-jsx))

**Comment by ryanfchase** [comment]
@entrotech -- requesting dev lead review, and then prioritization. Also, I never know what to do with the Stakeholder specific Action Items that come with the Blank Issue template. Let me know if these Action Items are necessary, happy to amend the ticket.

<details><summary>Stakeholder specific Action Items</summary>
<p>

- [ ] Once finalized, add before and after images to the slide in the staging deck for this issue (see link under resources)
- [ ] Get Stakeholder sign-off via the stakeholder meeting slide deck.

</p>
</details>

**Comment by Rabia2219** [comment]
@nchanyal Please provide update

<details><summary>Instructions</summary>

1. Progress: "What is the current status of your project? What have you completed and what is left to do?"
2. Blockers: "Difficulties or errors encountered."
3. Availability: "How much time will you have this week to work on this issue?"
4. ETA: "When do you expect this issue to be completed?"
5. Pictures (if necessary): "Add any pictures that will help illustrate what you are working on."
</details> 

You can use this template
```
1. Progress: 
2. Blockers: 
3. Availability:
4. ETA:
5. Pictures (if necessary): 
```
